model/trigger/interface.go

```diff
@@ -0,0 +1,49 @@
+package trigger
+
+import (
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/evergreen-ci/evergreen/model/notification"
+	"github.com/pkg/errors"
+)
+
+// `eventHandlerFactory`s create `eventHandler`s capable of validating triggers
+type eventHandlerFactory func() eventHandler
+
+// EventHandler
+type eventHandler interface {
```

Comment:
Having a private type but public methods feels weird. If the interface won't ever be used outside the package I would make everything private

Reply:
I don't believe golang supports private methods.